### DISCUSSION TITLE
How uploaded ticket layout mechanism work?

### DISCUSSION THREAD

**Discussion by Mike20403** [pretix/pretix#4186] [General]
I'm currently working on ticket layout to create a ticket layout , when I delete the ticket layout, I see that its background file which I uploaded is not being removed then. So I wonder, how the ticket background being process in this case when I delete it ? My expected behaviour is: the background pdf is being removed instantly.
And when I upload my background files through upload API from pretix, I see that it upload to this directory media/cachedfiles instead of the location where pretix does : data\media\pub\{organizerSlug}\{eventSlug}\ticketoutputpdf. 
Please help me 
![image](https://github.com/pretix/pretix/assets/57946382/dc3c9482-dedb-41d2-a54b-50c7a20d60e7)

**Comment by raphaelm** [comment]
I agree it should be removed, although there are some edge cases around it that make implementing this fully quite hard, as the upload does not immediately get saved
